test/model.js

```diff
@@ -428,27 +421,24 @@
   });
 
   QUnit.test("defaults", function(assert) {
-    assert.expect(4);
+    assert.expect(5);
     var Defaulted = Backbone.Model.extend({
       defaults: {
-        "one": 1,
-        "two": 2
+        one: 1,
+        two: 2
       }
     });
     var model = new Defaulted({two: undefined});
     assert.equal(model.get('one'), 1);
     assert.equal(model.get('two'), 2);
+    model = new Defaulted({two: 3});
+    assert.equal(model.get('one'), 1);
+    assert.equal(model.get('two'), 3);
     Defaulted = Backbone.Model.extend({
-      defaults: function() {
-        return {
-          "one": 3,
-          "two": 4
-        };
-      }
+      defaults: { hasOwnProperty: true, }
     });
-    model = new Defaulted({two: undefined});
```

Comment:
If we made it return a new instance instead of modifying the destination, it would work perfectly. We're kind of headed down that path with https://github.com/jashkenas/underscore/pull/2232.